### PR TITLE
Kill the GUI with an exception if the user hits CTRL-C

### PR DESCRIPTION
**Pull request by virtuald**
I need a windows user to try this out and tell me if something bad happens. Should be tested in conjunction with wpilib 2022.3.1.1.

Depends on:

* https://github.com/wpilibsuite/allwpilib/pull/3995
* https://github.com/wpilibsuite/allwpilib/pull/4034

### CODE REVIEW

File: halsim_gui/_ext/main.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+} // namespace wpi::gui
+
+RPYBUILD_PYBIND11_MODULE(m) {
+
+  m.def("_kill_on_signal", []() {
+    wpi::gui::AddLateExecute([] {
+      {
+        py::gil_scoped_acquire gil;
+        if (PyErr_CheckSignals() == -1) {
+          throw py::error_already_set();
```
Comment (virtuald) [inline halsim_gui/_ext/main.cpp]: @PeterJohnson if I just throw an exception on Windows/OSX from `wpi::gui::AddLateExecute`... anything bad going to happen? Seems to be fine on Linux. 

Eventually pybind11 will catch the exception and propagate it as a python exception to whomever called `HAL_RunMain`.

Comment (virtuald) [reply halsim_gui/_ext/main.cpp]: If I just call `wpi::gui::Exit`, the python error indicator is still set, and python becomes very unhappy when you pop back into python code with an active exception.

**Comment by TheTripleV** [comment]
Will this close https://github.com/robotpy/pyfrc/issues/174?

**Comment by virtuald** [comment]
It closes it on Linux, yes.

**Pull request merged**